make/autoconf/flags-cflags.m4

```diff
@@ -122,6 +122,8 @@ AC_DEFUN([FLAGS_SETUP_DEBUG_SYMBOLS],
             # Add debug prefix map gcc system include paths, as they cause
             # non-deterministic debug paths depending on gcc path location.
             DEBUG_PREFIX_MAP_GCC_INCLUDE_PATHS
+            DEBUG_PREFIX_CFLAGS="$DEBUG_PREFIX_CFLAGS -fdebug-prefix-map=\$(SUPPORT_OUTPUTDIR)/="
+            DEBUG_PREFIX_CFLAGS="$DEBUG_PREFIX_CFLAGS -fdebug-prefix-map=\$(HOTSPOT_OUTPUTDIR)/="
```

Comment:
This is only a problem if the build directory is located outside the workspace root. In the case where the build directory is inside the root, I think it would be preferable to stick to just the one prefix map, for consistent relative paths for all affected files. If the build directory is outside, like in your example, I would suggest adding just `$(OUTPUTDIR)/=` to both `-fdebug-prefix-map` and `-fmacro-prefix-map` for consistency. We don't seem to have a variable indicating if the OUTPUTDIR is inside WORKSPACE_ROOT or not, but we can add one in basic.m4 where the OUTPUTDIR is initialized. Then you can conditionally add the extra prefix maps based on that variable.

Reply:
I have not thought this fully through, but just throwing out a question: would this work even if one build directory were inside the workspace root, and the other not?

Reply:
> Oh you are right, we should always add the OUTPUTDIR to both maps regardless of if it's internal to the WORKSPACE_ROOT or not. Paths relative to the OUTPUTDIR should be stable across builds in different output directories, but paths inside the OUTPUTDIR, expressed as relative to the WORKSPACE_ROOT will vary.

@erikj79 @magicus Yes, it needs to always be present as you've pointed out the OUTPUTDIR when within the WORKSPACE will be eg.WORKSPACE/build/linux-aarch64-server-release/support and the standard WORKSPACE mapping will make that build/linux-aarch64-server-release/support, but a build dir outside will be simply <any_path>/support

Reply:
It seems correct to include the output dir in the remapping, but I have two objections/questions to the way you are doing it.
1) Why not just use OUTPUTDIR instead of the two specialized subdirs? That is simpler, more general and future-proof.
2) Why not expand the value of the OUTPUTDIR variable? I.e.  
```
 DEBUG_PREFIX_CFLAGS="$DEBUG_PREFIX_CFLAGS -fdebug-prefix-map=$OUTPUTDIR/="
 ```

instead of trying to preserve it as a variable to be expanded in the make execution.

Reply:
Also, this mapping business is getting really convoluted. :-( I did not like it as it was, and this patch makes it even worse. I think we need to rewrite this to create some kind of dict/map, and then iterate over the map and apply `-fdebug-prefix-map` to all items in the map. Unfortunately, the data structures available in shell scripts is limited and will make this a bit tricky to pull off. :(

Reply:
> It seems correct to include the output dir in the remapping, but I have two objections/questions to the way you are doing it.
> 
>     1. Why not just use OUTPUTDIR instead of the two specialized subdirs? That is simpler, more general and future-proof.
> 
>     2. Why not expand the value of the OUTPUTDIR variable? I.e.
> 
> 
> ```
>  DEBUG_PREFIX_CFLAGS="$DEBUG_PREFIX_CFLAGS -fdebug-prefix-map=$OUTPUTDIR/="
> ```
> 
> instead of trying to preserve it as a variable to be expanded in the make execution.

Ah thanks @magicus I hadn't seen that existed! that would simplify it, i'll try that now

Reply:
> Also, this mapping business is getting really convoluted. :-( I did not like it as it was, and this patch makes it even worse. I think we need to rewrite this to create some kind of dict/map, and then iterate over the map and apply `-fdebug-prefix-map` to all items in the map. Unfortunately, the data structures available in shell scripts is limited and will make this a bit tricky to pull off. :(

yeah, although that's not easy for example for the gcc include paths that we have to work out based on the users compiler paths, so it wouldn't be a simple dict

Reply:
@magicus thank you for the tip, it works nicely, ready for re-review please